core/embed/trezorhal/stm32f4/usb/usbd.c

```diff
@@ -149,6 +156,32 @@ void usb_init(const usb_dev_info_t *dev_info) {
   usb->next_iface_desc =
       (usb_interface_descriptor_t *)(usb->config_buf +
                                      usb->config_desc->wTotalLength);
+}
+
+void usb_deinit(void) {
+  usb_driver_t *usb = &g_usb_driver;
+
+  usb_stop();
+
+  // Set usb->dev_handle.dev_state to USBD_STATE_INITIALIZED
+  // Set usb->config_desc to NULL
+  memset(&usb, 0, sizeof(usb_driver_t));
+}
+
+void usb_start(void) {
```

Comment:
should we return (sec)bool?

Reply:
see this change 5c362db1cd8169f39a83b593480bd5ab66e39a9f

1. `usb_start()` and `usb_stop()` are now fallible, returning a `secbool` - in my opinion, drivers like this should not use `ensure` for error handling.
2. The return value is checked in all places where these functions are called.
3. Renamed the `usb` driver variable to `drv` to be more consistent with other drivers (maybe we should agree whether `drv` or `driver` is better - currently, I use both, but I'd like to change it to `drv` everywhere).
4. Renamed some `usbd_` files to `usb_` - files with the `usbd_` prefix are part of the STM USB driver, while files with the `usb_` prefix are ours.

Reply:
I'd prefer `drv` too, meaning is clear and its shorter